File: iOSversion-Objective-C/iOSversion-Objective-C/ViewController.m
//
//  ViewController.m
//  iOSversion-Objective-C
//
//  Created by タケル on 2018/1/24.
//  Copyright © 2018年 タケル. All rights reserved.
//

#import "ViewController.h"
#include <sys/utsname.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}
- (NSInteger)iPhoneModel:(NSString*)string {
    
    NSString *cutiPhone = [[self platformCode] stringByReplacingOccurrencesOfString:[[UIDevice currentDevice] model] withString:@""];
    
    NSRange search = [cutiPhone rangeOfString:@","];
    
    NSString *cutComma = [cutiPhone stringByReplacingCharactersInRange:NSMakeRange(0,search.location) withString:@""];
    
    NSString *iPhoneModel = [cutiPhone stringByReplacingOccurrencesOfString:cutComma withString:@""];
    
    return [iPhoneModel intValue];
}

- (NSString*) platformCode {
    struct utsname systemInfo;
    uname(&systemInfo);
    NSString* platform =  [NSString stringWithCString:systemInfo.machine encoding:NSUTF8StringEncoding];
    
    return platform;
}
- (BOOL) isSupport {
    // http://stackoverflow.com/questions/39564510/check-if-device-supports-uifeedbackgenerator-in-ios-10
    
    // Private API
    
    if ([self iPhoneModel:[self platformCode]] >= 9) {
        if ([[[UIDevice currentDevice] valueForKey:@"_feedbackSupportLevel"] intValue] == 2) {
            return YES;
        }
    }
    
    //    if ([UINotificationFeedbackGenerator class]) {
    //        return YES;
    //    }
    return NO;
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Appinfo/Appinfo/ViewController.swift
//
//  ViewController.swift
//  Appinfo
//
//  Created by タケル on 2018/5/10.
//  Copyright © 2018年 OBFirm. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let mainBundle = Bundle.main
        let identifier = mainBundle.bundleIdentifier
        let info = mainBundle.infoDictionary
        let bundleId = mainBundle.object(forInfoDictionaryKey: "CFBundleName")
        let version = mainBundle.object(forInfoDictionaryKey: "CFBundleShortVersionString")
        
        print("[identifier]:",identifier!)
        print("[info]:",info!)
        print("[bundleId]:",bundleId!)
        print("[version]:",version!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

